spec/index.md

```diff
@@ -5033,6 +5083,175 @@ S with
 
 ```
 
+#### IC Management Canister: Take snapshot
+
+Only the controllers of the given canister can take a snapshot.
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.callee = ic_principal
+M.method_name = 'take_snapshot'
+M.arg = candid(A)
+M.caller ∈ S.controllers[A.canister_id]
+
+Cycles_reserved = cycles_to_reserve(S, A.canister_id, S.compute_allocation[A.canister_id], S.memory_allocation[A.canister_id] + memory_usage_snapshot(New_snapshot), S.canisters[A.canister_id])
```

Comment:
```
S.memory_allocation[A.canister_id] + memory_usage_snapshot(New_snapshot)
```
the addition wouldn't make sense if memory allocation is fixed (i.e., not best-effort)

Reply:
We should pass S.memory_allocation[A.canister_id] here and add a new argument to cycles_to_reserve representing the snapshots in the IC state.

Reply:
Discussing offline.